### PR TITLE
replace community canton snapshot with our bazel-built canton deploy jar

### PR DESCRIPTION
**Pull request by paulbrauner-da**

### CODE REVIEW

File: test-common/canton/it-lib/src/main/com/daml/CantonFixture.scala [hunk 1]
```diff
@@ -60,15 +60,14 @@ trait CantonFixtureWithResource[A]
...
-    if (devMode) CantonRunner.cantonPatchPath else CantonRunner.cantonPath
+  protected lazy val cantonJar: Path = CantonRunner.cantonPath
 
   // This flag setup some behavior to ease debugging tests.
   //  If `true`
   //   - temporary file are not deleted (this requires "--test_tmpdir=/tmp/" or similar for bazel builds)
   //   - some debug info are logged.
   //   - output from the canton process is sent to stdout
-  protected val cantonFixtureDebugMode = false
+  protected val cantonFixtureDebugMode = true
```
Comment (remyhaemmerle-da) [inline test-common/canton/it-lib/src/main/com/daml/CantonFixture.scala:70]:
```suggestion
  protected val cantonFixtureDebugMode = false
```

File: compiler/damlc/tests/BUILD.bazel [hunk 2]
```diff
@@ -702,7 +702,6 @@ da_haskell_test(
         ":repl-test.dar",
         "//compiler/damlc",
         "//daml-script/daml:daml-script.dar",
-        "//test-common/canton:lib",
```
Comment (remyhaemmerle-da) [inline compiler/damlc/tests/BUILD.bazel:705]: What's not used ?

File: test-common/canton/it-lib/src/main/com/daml/CantonRunner.scala [hunk 3]
```diff
@@ -31,10 +31,7 @@ object CantonRunner {
   private[integrationtest] def toJson(s: String): String = JsString(s).toString()
   private[integrationtest] def toJson(path: Path): String = toJson(path.toString)
 
-  lazy val cantonPath =
-    Paths.get(rlocation("test-common/canton/canton_deploy.jar"))
-  lazy val cantonPatchPath =
-    Paths.get(rlocation("test-common/canton/canton-patched_deploy.jar"))
+  lazy val cantonPath = Paths.get(rlocation("canton/community_app_deploy.jar"))
```
Comment (remyhaemmerle-da) [inline test-common/canton/it-lib/src/main/com/daml/CantonRunner.scala:34]: :+1:

**Review by remyhaemmerle-da — APPROVED** [review]
Fantastic !

File: compiler/damlc/tests/BUILD.bazel (hunk 2 above)
Comment (paulbrauner-da) [reply compiler/damlc/tests/BUILD.bazel:705]: No it wasn't! The Canton sandbox is instantiated via a function provided by `//libs-haskell/test-utils`, which packs the jar already.

Comment (remyhaemmerle-da) [reply compiler/damlc/tests/BUILD.bazel:705]: :+1:

**Pull request merged**